misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -129,6 +136,21 @@ private IBundleCoverage analyzeStructure() throws IOException {
         return coverageBuilder.getBundle(title);
     }
 
+    private void createXMLReport(IBundleCoverage bundleCoverage) throws IOException {
+            XMLFormatter xmlFormatter = new XMLFormatter();
+            File reportFile = new File(target.getReportPath().resolve(
+                    this.module.moduleName().toString()).resolve(REPORT_XML_FILE).toString());
+            reportFile.getParentFile().mkdirs();
+            IReportVisitor visitor = xmlFormatter.createVisitor(new FileOutputStream(reportFile));
+            visitor.visitInfo(execFileLoader.getSessionInfoStore().getInfos(),
+                    execFileLoader.getExecutionDataStore().getContents());
+
+            visitor.visitBundle(bundleCoverage,
+                    new DirectorySourceFileLocator(null, "utf-8", TAB_WIDTH));
```

Comment:
lets the `StandardCharsets.UTF_8` instead of hardcoding the string

Reply:
Why the first param is passed as `null` ? Can we check whether we have another constructor to be used here?

Reply:
Yeah, don't have to pass a ```DirectorySourceFileLocator``` object. I passed null in place of the object.